### ISSUE TITLE
Verifytypes false-positive for contextmanager docstring

### ISSUE THREAD

**Issue by vemel** [microsoft/pyright#1581]
**Describe the bug**
Functions wrapped with `@contextlib.contextmanager` decorator always reported as missing docstring.

I believe, that it is a bug in `typeshed`, not in `pyright`, because if you replicate `contextlib.contextmanager`, functions are analyzed correctly.

**To Reproduce**
```python
from contextlib import contextmanager
from typing import Iterator

@contextmanager
def no_docstr() -> Iterator[int]:
    "pyright does not see this docstring!"
    yield 5
```

**Expected behavior**
Functions with docstrings should not be reported as undocumented.

**Screenshots or Code**

```python
# file: test_package/test.py
from contextlib import contextmanager, _GeneratorContextManager
from typing import Iterator
from functools import wraps

def mycontextmanager(func):
    "Exact copy of contextlib.contextmanager"
    @wraps(func)
    def helper(*args, **kwds):
        return _GeneratorContextManager(func, args, kwds)
    return helper

def docstr() -> Iterator[int]:
    "this docstring is checked correctly"
    yield 5

@contextmanager
def no_docstr() -> Iterator[int]:
    "pyright does not see this docstring!"
    yield 5

@mycontextmanager
def no_docstr2() -> Iterator[int]:
    "this one is visible as well, weird"
    yield 5
```

If you check `Symbol` for `no_docstr`, it does not have a docstring:

```javascript
{
  name: '',
  fullName: '',
  moduleName: '',
  flags: 32768,
  parameters: [
    {
      category: 1,
      name: 'args',
      type: [Object],
      hasDeclaredType: true
    },
    {
      category: 2,
      name: 'kwargs',
      type: [Object],
      hasDeclaredType: true
    }
  ],
  docString: undefined,
  declaredReturnType: {
    category: 8,
    classType: {
      category: 7,
      details: [Object],
      skipAbstractClassTest: false,
      flags: 1,
      typeArguments: [Array],
      isTypeArgumentExplicit: true,
      tupleTypeArguments: undefined
    },
    flags: 2
  },
  typeVarScopeId: '/home/vemel/pyright/packages/pyright/dist/typeshed-fallback/stdlib/contextlib.pyi.871'
}
```

So probably a solution would be to skip functions if `symbolType.details.name` is empty. This will not report undocumented context managers though.

**VS Code extension or command-line**

```bash
node pyright/packages/pyright --verifytypes test_package --ignoreexternal
node pyright/packages/pyright --verifytypes test_package
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

The issue is that the `contextmanager` decorator returns a different function from the one being decorated, and the new function has no docstring. I've modified the code to replicate the pre-decorated function's docstring into the post-decorated function.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.119, which I just published. It will also be included in the next version of Pylance.
